src/main/java/ladder/dto/LadderConsoleResult.java

```diff
@@ -0,0 +1,48 @@
+package ladder.dto;
+
+import ladder.*;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class LadderConsoleResult {
+
+    private final List<Member> members;
+    private final List<LadderLine> ladderLines;
+
+    private LadderConsoleResult(final List<Member> members, final List<LadderLine> ladderLines) {
+        this.members = new ArrayList<>(members);
+        this.ladderLines = new ArrayList<>(ladderLines);
+    }
+
+    public static LadderConsoleResult newInstance(final LadderGame ladderGame) {
+        return newInstance(ladderGame.getMembers(), ladderGame.getLadder());
+    }
+
+    public static LadderConsoleResult newInstance(final Members members, final Ladder ladder) {
+        return new LadderConsoleResult(members.getMembers(), ladder.getLines());
+    }
+
+    public List<Member> getMembers() {
+        return members;
```

Comment:
```suggestion
        return Collections.unmodifiablelist(members);
```

Collections.unmodifiablelist()를 활용하여 read-only로 리턴하여 외부에서 값의 변조를 막을 수 있습니다.

https://www.geeksforgeeks.org/collections-unmodifiablelist-method-in-java-with-examples/
https://stackoverflow.com/questions/51881801/why-is-it-better-to-return-unmodifiablelist-from-a-method-where-list-is-created

다만, 이 경우에도 List에 add, remove 등을 제한할 뿐 속성을 변화시키는 것을 제한하지는 않습니다.
깊은 복사를 위해 불변 객체를 활용하는 것은 어떨까요

> https://os94.tistory.com/153